### PR TITLE
EUI 3740 implementing generic filter

### PR DESCRIPTION
**Pull request by udaydenduluri33**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-3740?filter=-1

### Change description ###
Add generic filter to the the `task-home.component.ts`

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

**Review by kuda1992 — APPROVED** [review]
looks good

File: src/work-allocation-2/containers/task-home/task-home.component.html [hunk 1]
```diff
@@ -4,7 +4,24 @@ <h3 class="govuk-heading-xl">My work</h3>
...
+      <button class="govuk-button hmcts-button--secondary" type="button" (click)="toggleFilter = !toggleFilter">
+        <ng-container *ngIf="!toggleFilter">
+          Show filter
+        </ng-container>
+        <ng-container *ngIf="toggleFilter">
+          Hide filter
+        </ng-container>
+      </button>
+      <div class="govuk-!-padding-3 filter govuk-!-margin-bottom-3" *ngIf="toggleFilter"
+           style="background-color: #f2f2f1; width: 75%">
```
Comment (rayliangatos) [inline src/work-allocation-2/containers/task-home/task-home.component.html:16]: can this style move to css?

File: src/work-allocation-2/containers/task-home/task-home.component.html [hunk 2]
```diff
@@ -4,7 +4,24 @@ <h3 class="govuk-heading-xl">My work</h3>
...
+        <h3 class="govuk-heading-m">Immigration & Asylum</h3>
```
Comment (rayliangatos) [inline src/work-allocation-2/containers/task-home/task-home.component.html:17]: can this 'Immigration & Asylum' be dynamic or the name be more generic? What happened if more jurisditions enable work allocation feature.

Comment (kuda1992) [reply src/work-allocation-2/containers/task-home/task-home.component.html]: addressed

**Review by rayliangatos — APPROVED** [review]

**Pull request merged**